### DISCUSSION TITLE
How to access additional params during sign in?

### DISCUSSION THREAD

**Discussion by grantspilsbury** [nextauthjs/next-auth#5699] [Help]
### Question 💬

I am trying to access additional param during google sign in.

In my React component I have
`signIn('google', { callbackUrl }, { foo: 'bar' })`

In debug mode I see the additional param
```
[next-auth][debug][GET_AUTHORIZATION_URL] {
  url: 'https://accounts.google.com/o/oauth2/v2/auth?client_id=c1lvjn3.apps.googleusercontent.com&scope=openid%20email%20profile&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fapi%2Fauth%2Fcallback%2Fgoogle&foo=bar&state=tvfxXKcuu4jDqVo&code_challenge=j-dz&code_challenge_method=S256',
```

But how do I access these params?

If I log the values received by the signin callback I don't see the additional params. Where in the nextauth options should I be retrieving these additional params?

### How to reproduce ☕️

```
  System:
    OS: macOS 12.5
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 1.84 GB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.14.0 - ~/.nvm/versions/node/v16.14.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.14.0/bin/yarn
    npm: 8.13.2 - ~/.nvm/versions/node/v16.14.0/bin/npm
  Browsers:
    Chrome: 107.0.5304.87
    Edge: 107.0.1418.24
    Safari: 16.0
  npmPackages:
    next: 12.2.2 => 12.2.2 
    next-auth: ^4.12.3 => 4.15.0 
    react: 18.1.0 => 18.1.0
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by devkiran** [comment]
@grantspilsbury I think NextAuth appends the additional parameters to the authentication URL as a query string and the authentication provider do not return it after the authentication. Append it to the callback URL if you want to access the parameters after the authentication flow completes.

**Comment by grantspilsbury** [comment]
> 

Thanks @devkiran any idea how to do the same for email sign in?

**Comment by devkiran** [comment]
Can we do something like this?

```js
await signIn('email', {
  email: 'email@example.com',
  callbackUrl: '/callback?foo=bar',
});
```

You can access the param `foo` at page `/callback` after the authentication.
